core/store/src/flat/storage.rs

```diff
@@ -216,44 +234,87 @@ impl FlatStorage {
     /// Update the head of the flat storage, including updating the flat state in memory and on disk
     /// and updating the flat state to reflect the state at the new head. If updating to given head is not possible,
     /// returns an error.
-    pub fn update_flat_head(&self, new_head: &CryptoHash) -> Result<(), FlatStorageError> {
+    /// If `strict`, then unconditionally sets flat head to the given block.
+    /// If not `strict`, then uses `new_head` as a hint and may set flat head to
+    /// `new_head` or to a predecessor of `new_head`.
+    pub fn update_flat_head(
+        &self,
+        new_head: &CryptoHash,
```

Comment:
nit: rename it with just `block_hash` because this is not necessarily a new FS head.